### PR TITLE
Used new version of is-core library with better permissions system

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -8,9 +8,56 @@ To use django-fperms-iscore in a project, add it to your `INSTALLED_APPS`:
...
         ...
     )
 
-    # set the extended model for FPerms
-    PERM_MODEL = 'fperms_iscore.IsCorePerm'
+
+Cores
+-----
+
+Yyou can use `fperms_iscore.permissions.FPermPermission` with your cores to to set dynamic permissions system::
```
Comment (Formulka) [inline docs/usage.rst]: Yyou -> You

File: example/dj/apps/issue_tracker/tests/rest_permissions.py [hunk 2]
```diff
@@ -1,147 +1,93 @@
...
-    def test_user_can_update_only_itself(self):
-        user = self.get_user_obj()
-        resp = self.put('%s%s/' % (self.USER_API_URL, user.pk), data={})
-        assert_http_forbidden(resp)
+        logged_user = self.logged_user.user
 
-        user = self.logged_user.user
-        resp = self.put('%s%s/' % (self.USER_API_URL, user.pk), data={})
-        assert_valid_JSON_response(resp)
+        issue_read_permission = Perm.objects.get(codename='{}__{}'.format('issue', 'read'))
```
Comment (Formulka) [inline example/dj/apps/issue_tracker/tests/rest_permissions.py]: nebyl by na tohle fajn nějakej helper? se to tu opakuje

Comment (matllubos) [reply example/dj/apps/issue_tracker/tests/rest_permissions.py]: jen 4x, je to v testech, to bych neřešil, zhoršilo by to čitelnost

File: docs/usage.rst [hunk 3]
```diff
@@ -8,9 +8,57 @@ To use django-fperms-iscore in a project, add it to your `INSTALLED_APPS`:
...
+        permissions = PermissionsSet(
+            'create': FPermPermission('user__create', 'Create user'),
+            'read': FPermPermission('user__read', 'Read user'),
+            'update': FPermPermission('user__update', 'Update user'),
+            'delete': FPermPermission('user__delete', 'Delete user'),
+            'change_password': FPermPermission('user__change_password', register=False),
+        )
+
+
+As you can see class `fperms_iscore.permissions.FPermPermission` has two initial parameters. The first one is code name which is used for permission identification in the database and second one is a optional verbose name for better user experience. The last permission `change_password` has set parameter `register=False`, it means that permission will not be created but it will be used already defined permission with same code.
```
Comment (rubickcz) [inline docs/usage.rst]: - ... second one is **an** optional ...
- ... but it will **use** already defined ...

File: docs/usage.rst [hunk 4]
```diff
@@ -8,9 +8,57 @@ To use django-fperms-iscore in a project, add it to your `INSTALLED_APPS`:
...
+    from fperms_iscore.mixins.PermISCoreMixin
+    from fperms_iscore.main.PermUIRESTModelISCore
+
+    class UserISCore(PermISCoreMixin, UIRESTModelISCore):
+
+        model = User
+
+    class IssueISCore(PermUIRESTModelISCore):
+
+        model = User
```
Comment (rubickcz) [inline docs/usage.rst]: I think this should be `Issue`.

File: docs/usage.rst [hunk 5]
```diff
@@ -8,9 +8,57 @@ To use django-fperms-iscore in a project, add it to your `INSTALLED_APPS`:
...
+
+.. DANGER::
+   Superuser has always access to all operations. His access rights will not depend on the selected permissions or groups.
+
+Commands
+--------
+
+We right now have implemented cores with a dynamic permission check. But database permissions should be somehow stored inside a database. The command `sync_permissions` is there for this purpose. The command will create new permissions and updated already generated permissions.
```
Comment (rubickcz) [inline docs/usage.rst]: ... and **update*** already generated permissions.

File: example/dj/apps/issue_tracker/tests/factories.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+from factory import fuzzy
+
+from django.contrib.auth.models import User
+
+from issue_tracker.models import Issue
+
+
+class UserFactory(factory.DjangoModelFactory):
+
+    username = fuzzy.FuzzyText(length=10)
```
Comment (rubickcz) [inline example/dj/apps/issue_tracker/tests/factories.py]: TIP: You can use `factory.Faker('user_name')` to get more realistic username.

File: example/dj/apps/issue_tracker/tests/ui_permissions.py [hunk 7]
```diff
@@ -1,127 +1,103 @@
...
 
     @login(is_superuser=False)
-    def test_only_superuser_may_add_user(self):
-        USERNAME = 'new_nick'
-
-        resp = self.post('%sadd/' % self.USER_UI_URL, data={'add-is-user-username': USERNAME,
-                                                            'add-is-user-password': 'password'})
-        assert_http_forbidden(resp)
-        assert_false(User.objects.filter(username=USERNAME).exists())
+    def test_user_without_permission_should_do_nothing(self):
```
Comment (rubickcz) [inline example/dj/apps/issue_tracker/tests/ui_permissions.py]: The test with same method name already exists.

Comment (Formulka) [reply example/dj/apps/issue_tracker/tests/ui_permissions.py]: but it is in another testcase

File: fperms_iscore/permissions.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+            )
+        )
+        try:
+            return perms.get()
+        except Perm.DoesNotExist:
+            return None
+
+    def has_permission(self, name, request, view, obj=None):
+        """
+        Return `True` if user has set permission of is superuser
```
Comment (Formulka) [inline fperms_iscore/permissions.py]: **or** is a superuser

File: example/dj/apps/issue_tracker/tests/commands.py [hunk 9]
```diff
@@ -1,30 +1,25 @@
...
-        Token.objects.filter(pk__in=[token.pk for token in expired_tokens]).update(
-            last_access=timezone.now() - timedelta(seconds=settings.AUTH_MAX_TOKEN_AGE))
-        call_command('clean_tokens', stdout=StringIO(), stderr=StringIO())
-        assert_equal(Token.objects.filter(pk__in=[token.pk for token in not_expired_tokens]).count(),
-                     settings.AUTH_COUNT_USER_PRESERVED_TOKENS - 5)
-        assert_equal(Token.objects.filter(pk__in=[token.pk for token in expired_tokens]).count(), 5)
+        for model_name in model_names:
+            for perm_name in perm_names:
+                assert_equal(
+                    Perm.objects.filter(type='is_core', codename='{}__{}'.format(model_name, perm_name)).count(), 1
```
Comment (Formulka) [inline example/dj/apps/issue_tracker/tests/commands.py]: `type=enums.PERM_TYPE_CORE`

**Comment by matllubos** [comment]
waiting for version

**Pull request merged**